plugins/action_send_notification/src/sendNotification.ts

```diff
@@ -0,0 +1,127 @@
+import { ActionQueueStatus } from '../../../src/generated/graphql'
+import { ActionPluginType } from '../../types'
+import databaseMethods from './databaseMethods'
+import path from 'path'
+import nodemailer from 'nodemailer'
+import marked from 'marked'
+import config from '../config.json'
+import { Attachment } from 'nodemailer/lib/mailer'
+import { getFilePath } from '../../../src/components/files/fileHandler'
+import { ActionApplicationData } from '../../../src/types'
+
+const sendNotification: ActionPluginType = async ({ parameters, applicationData, DBConnect }) => {
+  const db = databaseMethods(DBConnect)
+  const { host, port, secure, user, defaultFromName, defaultFromEmail } = config
+  const {
+    userId = applicationData?.userId,
+    email = applicationData?.email,
+    fromName = defaultFromName,
+    fromEmail = defaultFromEmail,
+    subject,
+    message,
+    attachments = [],
+    sendEmail = true,
+  } = parameters
+
+  const {
+    environmentData: { appRootFolder, filesFolder },
+  } = applicationData as ActionApplicationData
+
+  const transporter = nodemailer.createTransport({
+    host,
+    port,
+    secure,
+    auth: {
+      user,
+      pass: process.env.SMTP_PASSWORD,
+    },
+  })
+
+  try {
+    const emailAddressString = stringifyEmailRecipientsList(email)
+
+    // Create notification database record
+    console.log('Creating notification...')
+    let notificationResult
+    notificationResult = await db.createNotification({
+      userId,
+      applicationId: applicationData?.applicationId,
+      emailAddressString,
+      subject,
+      message,
+      attachments,
+    })
+
```

Comment:
Should check or the try catch would stop here if notification not correctly created?

Reply:
If the database method throws an error, the Action will escape from the "Try" block and run "catch" -- is that what you meant?

Reply:
Yes, I was unsure if the notification could not get created and not errors thrown, but I think it would throw because of the return is accessing the row[0]

Reply:
> > * I could make the `sendMail` part of the Action itself async. That would just mean that there would be no confirmation that the email actually sent as part of the Outcome object (it would just show the notification record info), but there would be a record in the "email_sent" field of the notification table.
> 
> I vote for this solution

Yeah, I think that's probably best -- will do it now.

Reply:
Righto -- that's done now, you shouldn't get the timeout on Join Org now: d35dd23b